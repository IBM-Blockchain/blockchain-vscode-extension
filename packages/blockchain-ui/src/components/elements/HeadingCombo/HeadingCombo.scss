@import '@carbon/colors/scss/colors';
@import '@carbon/layout/scss/_spacing';

.heading-combo-container {
    h3 {
        margin-bottom: $spacing-05;
        line-height: inherit;
        font-size: 1.25rem;
        font-weight: 900;
        color: inherit;
    }

    p {
        margin-bottom: $spacing-06;
        color: $gray-70;
        color: inherit;
    }
}
