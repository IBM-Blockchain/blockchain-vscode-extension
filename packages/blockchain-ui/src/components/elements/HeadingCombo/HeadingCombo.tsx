import React from 'react';
import './HeadingCombo.scss';

interface IProps {
    comboStyle?: string;
    headingText: string;
    headingStyle?: string;
    subheadingText: string;
    subheadingStyle?: string;
}

function HeadingCombo(props: IProps): JSX.Element {
    let style: string = 'heading-combo-container';
    style += props.comboStyle ? ` ${props.comboStyle}` : '';

    return (
        <div className={style}>
            <h3 className={props.headingStyle}>{props.headingText}</h3>
            <p className={props.subheadingStyle}>{props.subheadingText}</p>
        </div>
    );
}

export default HeadingCombo;
