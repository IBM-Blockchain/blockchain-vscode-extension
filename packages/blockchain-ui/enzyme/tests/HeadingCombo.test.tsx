import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import HeadingCombo from '../../src/components/elements/HeadingCombo/HeadingCombo';

chai.should();
chai.use(sinonChai);

describe('HeadingCombo component', () => {
    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<HeadingCombo headingText='Here Is A Heading' subheadingText='And here is a subheading'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should apply any additional styles that are provided', () => {
        const component: any = mount (
            <HeadingCombo
                headingText='Here Is A Heading'
                subheadingText='And here is a subheading'
                comboStyle='extra-style-1'
                headingStyle='extra-style-2'
                subheadingStyle='extra-style-3'
            />
        );
        component.find('div').hasClass('extra-style-1').should.equal(true);
        component.find('h3').hasClass('extra-style-2').should.equal(true);
        component.find('p').hasClass('extra-style-3').should.equal(true);
    });
});
