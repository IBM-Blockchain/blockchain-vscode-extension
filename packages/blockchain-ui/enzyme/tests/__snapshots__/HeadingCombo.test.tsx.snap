// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingCombo component should render the expected snapshot 1`] = `
<div
  className="heading-combo-container"
>
  <h3>
    Here Is A Heading
  </h3>
  <p>
    And here is a subheading
  </p>
</div>
`;
